x-pack/plugins/code/public/reducers/symbol.ts

```diff
@@ -73,11 +73,12 @@ const generateStructureTree: (symbols: SymbolInformation[]) => any = symbols =>
     if (result) {
       return result;
     } else {
+      // TODO: Use Array.flat once supported
       const subTree = tree
         .filter(s => s.members)
         .map(s => s.members)
-        .flat();
-      if (subTree.length > 0) {
+        .reduce((prev, current) => (prev || []).concat(current || []), []);
```

Comment:
This can be simplified to the following since `prev` can never be falsy:

```js
.reduce((prev, current = []) => prev.concat(current), []);
```

Reply:
Also, since a `filter` is occurring based on truthiness, I don't think you need a default value for `current` either:

```js
.reduce((prev, current) => prev.concat(current), []);
```

Reply:
Also (again lol), you can avoid iterating over this collection 3 times by using a single reduce:

```js
const subTree = tree
  .reduce((subTree, tree) => tree.members ? subTree.concat(tree.members) : subTree, []);
```